csvjoin/functions.py

```diff
@@ -0,0 +1,257 @@
+    """
+        Finds the index of the column with the given name.
+
+        :param header: The header line of the CSV file.
+        :param column_name: The name of the column to find its index.
+        :type header: str
+        :type column_name: str
+        :return: The index of the column.
+        :rtype: int
+    """
+    header = header.rstrip()
+    arr = header.split(',')
+    try:
+        idx = arr.index(column_name)
+    except ValueError:
+        raise ValueError(f'The name {column_name} does not exist in one of the files!')
+
+    return idx
+
+
+def parse_data_line(line1: str, line2: str, column_index1: int, column_index2: int):
+    """
+        Joins and prints the two lines from CSV files.
+
+        :param line1: The first line to join.
+        :param line2: The second line to join.
+        :param column_index1: The index of the joining column in the first line.
+        :param column_index2: The index of the joining column in the second line.
+        :type line1: str
+        :type line2: str
+        :type column_index1: int
+        :type column_index2: int
+    """
+    new_line = str(line1)
+    new_line = new_line.rstrip()
+    new_line = new_line.split(',')
+
+    new_line2 = str(line2)
+    new_line2 = new_line2.rstrip()
+    new_line2 = new_line2.split(',')
+
+    if column_index2 != -1:
+        new_line2.pop(column_index2)
+    else:
+        new_line.pop(column_index1)
+
+    new_line.extend(new_line2)
+
+    line_to_print = ''
+    for _ in range(len(new_line)):
+        line_to_print += '{: >13} '
+
+    print(line_to_print.format(*new_line))
+
+
+def split_and_strip(line: str) -> [str]:
+    """
+        Copies the string, deletes the \n sign and
+        splits by comma.
+
+        :param line: The line to split and rstrip.
+        :type line: str
+        :return: The array of strings.
+        :rtype: [str]
+   """
+    new_line = str(line)
+    new_line = new_line.rstrip()
+    return new_line.split(',')
+
+
+def print_non_inner_join(file, header: str, taken_rows: set, join_type: str, column_index: int):
+    """
+        Prints non-inner type of join in order to "fill" NULL values.
+
+        :param file: The handle of the file.
+        :param header: The header of the second file to get the
+        number of the NULLs.
+        :param taken_rows: The set of already printed rows of the file.
+        :param join_type: The type of the join.
+        :param column_index: The column_index to pass it
+        to the parse_data_line function.
+        :type file: TextIO
+        :type header: str
+        :type taken_rows: set
+        :type join_type: str
+        :type column_index: int
+    """
+    file.seek(0, 0)
+    file.readline()
+    line = file.readline()
+
+    # creating NULL line string
+    header = header.split(',')
+    n = len(header)
+    filler = ''
+    for _ in range(n - 1):
+        filler += 'NULL'
+        filler += ','
+    filler += 'NULL'
```

Comment:
1. I don't understand why is this needed at all
2. Single-letter names are bad, use e.g. `header_length`
3. Use list comprehension with join instead of string concatenation